validation/testutils/mock_ethclient.go

```diff
@@ -0,0 +1,29 @@
+package testutils
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// MockEthClient is a mock that implements EthCaller for testing.
+type MockEthClient struct {
+	// Setup response maps keyed by method 4-byte signature + address, or
+	// just by method signature if that's simpler for your tests.
+	Responses map[string][]byte
+	Err       error
+}
```

Comment:
This can be reused by many of the validation unit tests so I extracted it into this `testutils` package where we can put other shared test mocks/helpers